src/lib/logging.ts

```diff
@@ -0,0 +1,453 @@
+import moment from 'moment';
+import { Platform, Plugin } from 'obsidian';
+/*
+ * EventEmitter2 is an implementation of the EventEmitter module found in Node.js.
+ * In addition to having a better benchmark performance than EventEmitter and being
+ * browser-compatible, it also extends the interface of EventEmitter with many
+ * additional non-breaking features.
+ *
+ * This has been added as EventEmitter in Node.JS is not available in the browser.
+ * https://www.npmjs.com/package/eventemitter2
+ */
+import { EventEmitter2 } from 'eventemitter2';
+
+/**
+ * All possible log levels
+ * @public
+ */
+export interface ILogLevel {
+    1: 'trace';
+    2: 'debug';
+    3: 'info';
+    4: 'warn';
+    5: 'error';
+}
+
+/**
+ * Logger class to handle consistency of logs across the plugin.
```

Comment:
This comment appears on LogManager too.

Is that intentional?
And consistency with what?

Reply:
The log entries when they are written to the log all have time stamps, indication of level and an option tracking ID so you can see calls between functions. This is used to correlate the queries from the query request to render.